.github/workflows/tests-smoke.yml

```diff
@@ -0,0 +1,35 @@
+name: Test [smoke]
+
+on: pull_request
+
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  test-matrix:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - uses: gradle/gradle-build-action@v2
+        env:
+          ORG_GRADLE_PROJECT_org.jetbrains.dokka.javaToolchain.test: 11
+        with:
+          gradle-home-cache-cleanup: true
+      - name: Run tests under Ubuntu/Macos
+        if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
+        run: ./gradlew clean test --stacktrace
+      - name: Run tests under Windows
+        if: matrix.os == 'windows-latest'
+        # Running tests without additional flags on windows agents leads to some very strange
+        # JVM crashes for some reason. Most likely a problem of Gradle/GitHub/Windows server
+        run: ./gradlew clean test --stacktrace --no-daemon --parallel --max-workers=1
```

Comment:
`--parallel` is the default, you want `--no-parallel`?

Reply:
No idea :( As I [pointed out](https://github.com/Kotlin/dokka/pull/2951#discussion_r1153718954), this was copy-pasted from the previous configuration. Without it, the windows tests are consistently failing due to JVM crashes. Unfortunately, there was no explanation in the commit/PR that introduced these changes :man_shrugging:

Reply:
Need `--no-daemon` to disable daemon for Windows, I patched this for Ktlint before, and `--parallel` is the default, could be removed for now.

Reply:
Looks like the tests passed, thank you!

Reply:
There are more flaky test failures with Daemon enabled on Windows.